serving/docker/partition/sm_neo_trt_llm_partition.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+#
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import json
+import logging
+import os
+import sys
+import traceback
+
+from utils import load_properties
+from tensorrt_llm_toolkit import create_model_repo
+
+# TODO: Merge the functionality of this file into trt_llm_partition.py
+# so all TRT-LLM partitioning is unified
+
+class InputConfiguration(Exception):
+    """Raise when Neo interface expectation is not met"""
+
+
+class CompilationFatalError(Exception):
+    """Raise for errors encountered during the TensorRT-LLM build process"""
+
+
+def write_error_to_file(error_message, error_file):
+    """
+    Write error messages to error file
+    """
+    try:
+        with open(error_file, "w", encoding="utf-8") as f:
+            json.dump({"error": repr(error_message)}, f)
+    except:
+        tb_exc = traceback.format_exc()
+        logging.error(f"Failed to write error file: {tb_exc}")
+
+
+def get_neo_env_vars():
+    """
+    Get environment variables required by the Neo interface
+    """
+    try:
+        compiler_options = os.environ.get("COMPILER_OPTIONS")
```

Comment:
No, `COMPILER_OPTIONS` should always be passed in to the container like the other flags, since this is how the customers specify using TRT-LLM for Neo in the first place. I will change it to match the others